### ISSUE TITLE
Fix `--controls` pre-filtering and loading in the `exec` and add `--tags`

### ISSUE THREAD

**Issue by aaronlippold** [mitre/inspec#1]
@scornwell had he had a possible solution worked out or at least thought you did - @earonne and I were talking and we thing this use case - aka only adding the listed control to the 'runner' on the back-end prior to profile execution .  

https://github.com/inspec/inspec/issues/2857 

and connect to this 

https://github.com/inspec/inspec/issues/1677 

I will also reopen this issue. But my thought is that once we figure out one the second would follow very very easily. Lastly, @tvaughen also has some good examples in his spec about adding specific controls to the inspec.runner object and exec'ing it.

All controls in profile evaluated when controls are selected using --controls · Issue #2857 · inspec/inspec
:control_knobs: Description Inspec cli allows for selection of controls using the following tag --controls=one two three A list of controls to run. Ignore all other tests. However inspec runner still evals all t...

Add a --tag and --tags flag which operates just like the --controls flag does. · Issue #1677 · inspec/inspec
Description I would like to be able to pass a --tag or --tags flag to the InSpec CLI to instruct a profile to only execute controls with the provided tag or tags. Use Case Assume I have a profile o...

These fixes are REALLY in the CORRECTNESS column in my mind - aka when we pass --controls that is REALLY does only exec those controls ...

**Comment by mschmeitz** [comment]
I am in favor of adding the '--tags' I also opt for adding --impact.

The description below is an implementation of the use case: See the results of must-compliant controls and should-compliant controls 

Use case:
Our devops teams use pipelines in which they must comply to all security controls (tagged with 'security'), so the pipeline must be able to execute all profiles but only the tests that are tagged with security.

The same profiles are used in running/production environment. So at the start of a release, the pipeline made sure the systems are compliant.
Since the level of security is increasing day by day, new controls are being added. The development pipeline takes these new demands into account from day 1 when these controls are released so new developments are compliant, however, in an Enterprise environment the remediation of all systems may take some time for lower priority risk mitigations. 
So it is required to define these controls on impact level Low, and only test security compliancy on impact level  Medium or higher, while for devops the complete control set can be used.

**Comment by aaronlippold** [comment]
yup - I agree with the use cases - and fundamentally would also help us
really streamline the use of the technology. We are looking for a group of
folks who want to help bring a PR for this into reality.
--------
Aaron Lippold

lippold@gmail.com

260-255-4779

twitter/aim/yahoo,etc.
'aaronlippold'

**Comment by Amndeep7** [comment]
Current situation is the controls filter now occurs before some of the evaluation so speeds are improved, but they still have issues with control files still needing to be evaluated/processed in some manner due to the fact that these files are not pure inspec dsl but also include ruby code.

The filter for tags was officially declined as additional functionality unless we submit (and get merged) a pr.

**Comment by camdenmoors** [comment]
Seems to be implemented as of https://github.com/inspec/inspec/pull/5643

**Comment by camdenmoors** [comment]
Will re-close after validation both of these issues have been addressed.
